op-service/sources/prefetching_eth_client_test.go

```diff
@@ -0,0 +1,169 @@
+package sources
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+// Define a test type for each method
+type testFunction func(t *testing.T, ctx context.Context, client *PrefetchingEthClient, mock *testutils.MockEthClient) error
```

Comment:
you pass `t` to the testFunction and make individual assertions inside, so you probably won't ever need to let it return an error